clang/lib/Driver/Driver.cpp

```diff
@@ -4134,9 +4063,33 @@ class OffloadingActionBuilder final {
           // FPGA aoco does not go through the link, everything else does.
           if (Input->getType() == types::TY_FPGA_AOCO)
             DeviceLibObjects.push_back(Input);
-          else
+          // FPGA aocr/aocx does not go through the link and is passed
+          // directly to the backend compilation step (aocr) or wrapped (aocx)
+          else if (types::isFPGA(Input->getType())) {
+            Action *AOCAction;
```

Comment:
Maybe something like:
```suggestion
            Action *FPGAAOTAction;
```
Not sure about the exact naming, but something that would untie a variable name from a particular BE tool and focus on the nature of the action instead.

Reply:
Done.